package-scripts/datadog-agent/preinst

```diff
@@ -34,10 +34,18 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
 
     # Prepare the GPG keys rotation:
     # Add the new one to the list of trusted APT keys.
+    # Some servers are on severely resticted networks.  Check for proxy,
+    # and check if key already exists.
     echo "Prepare Datadog Agent keys rotation"
     echo -n "  Add the new 'Datadog, Inc <package@datadoghq.com>' key to the list of APT trusted keys."
-    apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 382E94DE >/dev/null 2>&1 && echo " ... OK" || echo " ... failed"
-
+    if $( apt-key finger | grep -q 'A292 3DFF 56ED A6E7 6E55  E492 D3A8 0E30 382E 94DE' ); then
+        echo "... key already installed"
+    else
+        if [ "$http_proxy" != "" ]; then
+            HKP_PROXY="--keyserver-options http-proxy=$http_proxy"
```

Comment:
From `man gpg` in the `--keyserver-options` section:
> http-proxy=value
>     Set the proxy to use for HTTP and HKP keyservers.  This overrides the "http_proxy" environment variable, if any.

So I don't think we need to handle the proxy setting manually here.

I've also tried on a local environment and `apt-key adv` seems to pick up the proxy defined in the `http_proxy` env var on its own.

If you agree let's remove this block